conda_libmamba_solver/solver.py

```diff
@@ -738,3 +754,81 @@ def _called_from_conda_build(self):
             # frame[3] contains the name of the function in that frame of the stack
             and {"install_actions", "get_install_actions"} <= {frame[3] for frame in stack()}
         )
+
+    def _notify_conda_outdated(
+        self,
+        link_precs,
+        index: LibMambaIndexHelper = None,
+        final_state: Iterable[PackageRecord] = None,
+    ):
+        """
+        We are overriding the base class implementation, which gets called in
+        Solver.solve_for_diff() once 'link_precs' is available. However, we
+        are going to call it before (in .solve_final_state(), right after the solve).
+        That way we can reuse the IndexHelper and SolverOutputState instances we have
+        around, which contains the channel and env information we need, before losing them.
+        """
+        if index is None and final_state is None:
+            # The parent class 'Solver.solve_for_diff()' method will call this method again
+            # with only 'link_precs' as the argument, because that's the original method signature.
+            # We have added two optional kwargs (index and final_state) so we can call this method
+            # earlier, in .solve_final_state(), while we still have access to the index helper
+            # (which allows us to query the available packages in the channels quickly, without
+            # reloading the channels with conda) and the final_state (which gives the list of
+            # packages to be installed). So, if both index and final_state are None, we return
+            # because that means that the method is being called from .solve_for_diff() and at
+            # that point we will have already called it from .solve_for_state().
+            return
+        if not context.notify_outdated_conda or context.quiet:
+            # This check can be silenced with a specific option in the context or in quiet mode
+            return
+
+        current_conda_prefix_rec = PrefixData(context.conda_prefix).get("conda", None)
+        if not current_conda_prefix_rec:
+            # We are checking whether conda can be found in the environment conda is
+            # running from. Unless something is really wrong, this should never happen.
+            return
+
+        channel_name = current_conda_prefix_rec.channel.canonical_name
+        if channel_name == UNKNOWN_CHANNEL:
+            channel_name = "defaults"
+
+        # only check the loaded index if it contains the channel conda should come from
+        # otherwise ignore
+        index_channels = set(getattr(chn, "canonical_name", chn) for chn in index._channels)
+        if channel_name in index_channels:
+            return
```

Comment:
@jezdez I added this extra bit now. I hope that's ok!

Reply:
Yeah, that seems sensible.